fluss-spark/fluss-spark-common/src/main/scala/org/apache/fluss/spark/row/SparkAsFlussArray.scala

```diff
@@ -0,0 +1,131 @@
+
+  override def toByteArray: Array[Byte] = arrayData.toByteArray()
+
+  override def toShortArray: Array[Short] = arrayData.toShortArray()
+
+  override def toIntArray: Array[Int] = arrayData.toIntArray()
+
+  override def toLongArray: Array[Long] = arrayData.toLongArray()
+
+  override def toFloatArray: Array[Float] = arrayData.toFloatArray()
+
+  override def toDoubleArray: Array[Double] = arrayData.toDoubleArray()
+
+  /** Returns true if the element is null at the given position. */
+  override def isNullAt(pos: Int): Boolean = arrayData.isNullAt(pos)
+
+  /** Returns the boolean value at the given position. */
+  override def getBoolean(pos: Int): Boolean = arrayData.getBoolean(pos)
+
+  /** Returns the byte value at the given position. */
+  override def getByte(pos: Int): Byte = arrayData.getByte(pos)
+
+  /** Returns the short value at the given position. */
+  override def getShort(pos: Int): Short = arrayData.getShort(pos)
+
+  /** Returns the integer value at the given position. */
+  override def getInt(pos: Int): Int = arrayData.getInt(pos)
+
+  /** Returns the long value at the given position. */
+  override def getLong(pos: Int): Long = arrayData.getLong(pos)
+
+  /** Returns the float value at the given position. */
+  override def getFloat(pos: Int): Float = arrayData.getFloat(pos)
+
+  /** Returns the double value at the given position. */
+  override def getDouble(pos: Int): Double = arrayData.getDouble(pos)
+
+  /** Returns the string value at the given position with fixed length. */
+  override def getChar(pos: Int, length: Int): BinaryString =
+    BinaryString.fromBytes(arrayData.getUTF8String(pos).getBytes)
+
+  /** Returns the string value at the given position. */
+  override def getString(pos: Int): BinaryString =
+    BinaryString.fromBytes(arrayData.getUTF8String(pos).getBytes)
+
+  /**
+   * Returns the decimal value at the given position.
+   *
+   * <p>The precision and scale are required to determine whether the decimal value was stored in a
+   * compact representation (see {@link Decimal}).
+   */
+  override def getDecimal(pos: Int, precision: Int, scale: Int): Decimal = {
+    val sparkDecimal = arrayData.getDecimal(pos, precision, scale)
+    if (sparkDecimal.precision <= org.apache.spark.sql.types.Decimal.MAX_LONG_DIGITS)
+      Decimal.fromUnscaledLong(
+        sparkDecimal.toUnscaledLong,
+        sparkDecimal.precision,
+        sparkDecimal.scale)
+    else
+      Decimal.fromBigDecimal(
+        sparkDecimal.toJavaBigDecimal,
+        sparkDecimal.precision,
+        sparkDecimal.scale)
+  }
+
+  /**
+   * Returns the timestamp value at the given position.
+   *
+   * <p>The precision is required to determine whether the timestamp value was stored in a compact
+   * representation (see {@link TimestampNtz}).
+   */
+  override def getTimestampNtz(pos: Int, precision: Int): TimestampNtz =
+    TimestampNtz.fromMillis(SparkDateTimeUtils.microsToMillis(arrayData.getLong(pos)))
```

Comment:
I think we can introduce a method `TimestampNtz.fromMicros` like `TimestampLtz.fromEpochMicros` to suppoert convert from micro to `TimestampNtz`. Converting from micro to millis will lose the nano precisions.